server/grpc_service.go

```diff
@@ -1266,10 +1266,24 @@ func (s *GrpcServer) GetRegion(ctx context.Context, request *pdpb.GetRegionReque
 	if rc == nil {
 		return &pdpb.GetRegionResponse{Header: s.notBootstrappedHeader()}, nil
 	}
-	region := rc.GetRegionByKey(request.GetRegionKey())
+	var region *core.RegionInfo
+	// allow region miss temporarily if this key can't be found in the region tree.
+retryLoop:
+	for retry := 0; retry <= 10; retry++ {
+		region = rc.GetRegionByKey(request.GetRegionKey())
+		if region != nil {
+			break retryLoop
+		}
+		select {
+		case <-ctx.Done():
+			break retryLoop
+		case <-time.After(10 * time.Millisecond):
+		}
+	}
```

Comment:
How about abstracting it?

Reply:
not fix in this pr , will add in other pr.